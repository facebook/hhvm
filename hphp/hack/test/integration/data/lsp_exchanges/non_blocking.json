[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
      "initializeOptions": {},
      "processId": null,
      "rootPath": "${root_path}",
      "capabilities": {}
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/definition",
    "id": 2,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 7,
        "character": 11
      }
    }
  },
  {
    "comment": "remove hh_loop_forever() invocation to break the infinite loop",
    "jsonrpc": "2.0",
    "method": "textDocument/didOpen",
    "params": {
      "textDocument": {
        "uri": "${root_path}/__hh_loop_forever_foo.php",
        "languageId": "hack",
        "version": 1,
        "text": "<?hh //strict\n\nfunction __hh_loop_forever_foo(): int { return 4; }"
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "method": "shutdown",
    "params": {}
  }
]
