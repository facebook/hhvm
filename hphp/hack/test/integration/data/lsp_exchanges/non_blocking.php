<?hh // strict

function non_blocking_definition(): int {
  return 4;
}

function non_blocking(): int {
  return non_blocking_definition();
}
