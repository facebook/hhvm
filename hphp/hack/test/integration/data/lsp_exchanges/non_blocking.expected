[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": [
      {
        "uri": "file://${root_path}${path_sep}non_blocking.php",
        "range": {
          "start": {
            "line": 2,
            "character": 9
          },
          "end": {
            "line": 2,
            "character": 32
          }
        },
        "title": "non_blocking_definition"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "result": null
  }
]